client/components/marketing-survey/cancel-purchase-form/radio-option.jsx

```diff
@@ -46,3 +47,43 @@ export const radioOption = (
 		</FormLabel>
 	);
 };
+
+export const radioSelectOption = (
+	key,
+	radioValue,
+	onRadioChange,
+	onSelectChange,
+	radioPrompt,
+	selectLabel,
+	selectOptions,
+	selected
+) => {
+	const selectInputKey = `${ key }Select`;
+
+	const selectInput = selectOptions?.length && (
+		<FormLabel key={ selectInputKey } className="cancel-purchase-form__reason-select">
+			{ selectLabel }
+			<SelectDropdown
+				initialSelected={ selected }
+				options={ selectOptions }
+				onSelect={ onSelectChange }
+				compact
+			/>
+		</FormLabel>
+	);
+
+	return (
+		<React.Fragment key={ `fragment${ key }` }>
```

Comment:
Maybe we could set the key in the parent component, when we instantiate the select options. Usually component shouldn't need to be aware of the existence of other components around them.